src/Workspaces/Core/Portable/CodeGeneration/CodeGenerationHelpers.cs

```diff
@@ -38,17 +37,11 @@ internal static class CodeGenerationHelpers
             return factory.ThrowStatement(exceptionCreationExpression);
         }
 
-        public static TSyntaxNode AddAnnotationsTo<TSyntaxNode>(ISymbol symbol, TSyntaxNode syntax)
-            where TSyntaxNode : SyntaxNode
-        {
-            if (syntax != null && symbol is CodeGenerationSymbol)
-            {
-                return syntax.WithAdditionalAnnotations(
-                    ((CodeGenerationSymbol)symbol).GetAnnotations());
-            }
-
-            return syntax;
-        }
+        [return: NotNullIfNotNull("syntax")]
+        public static TSyntaxNode? AddAnnotationsTo<TSyntaxNode>(ISymbol symbol, TSyntaxNode? syntax) where TSyntaxNode : SyntaxNode
+            => symbol is CodeGenerationSymbol codeGenerationSymbol
+                ? syntax?.WithAdditionalAnnotations(codeGenerationSymbol.GetAnnotations())
+                : syntax;
 
         public static TSyntaxNode AddFormatterAndCodeGeneratorAnnotationsTo<TSyntaxNode>(TSyntaxNode node) where TSyntaxNode : SyntaxNode
             => node.WithAdditionalAnnotations(Formatter.Annotation, CodeGenerator.Annotation);
```

Comment:
The `CheckNodeType` just under this should accept null.

Reply:
that method isn't even used.  I've removed it :)